iroh-resolver/src/resolver.rs

```diff
@@ -1090,6 +1103,12 @@ impl<T: ContentLoader> Resolver<T> {
             let mut resolved_path = vec![cid];
 
             for part in tail {
+                // TODO(ramfox): we currently add an empty string to the path.tail vec
+                // when the original path ends in a "/". This is causing errors because
+                // we attempt to index into the data using an empty string, which fails
```

Comment:
@Arqu would love some feedback on this

i originally took out these lines:
https://github.com/n0-computer/iroh/blob/96c614839127a0b881ac100e3102abca70cb3a87/iroh-resolver/src/resolver.rs#L253-L255

But it looks like this was added for gateway compliance. Just want to confirm, because it's makin' the resolve code smelly :)

Reply:
Ok so the reason it's there is purely aesthetically (and compliance with spec) to redirect to a path with a trailing `/` if the result is a unixfs dir. For all intents and purposes you should basically use tail.filter(|s| !s.is_empty()) internally. It's just a way to pass info around it's a dir without having resolved the node yet.